doc/src/guides/solving/solve-numerically.md

```diff
@@ -0,0 +1,225 @@
+>>> x1 = Symbol('x1')
+>>> x2 = Symbol('x2')
+>>> f1 = 3 * x1**2 - 2 * x2**2 - 1
+>>> f2 = x1**2 - 2 * x1 + x2**2 + 2 * x2 - 8
+>>> print(nsolve((f1, f2), (x1, x2), (-1, 1)))
+Matrix([[-1.19287309935246], [1.27844411169911]])
+```
+
+### Solve overdetermined systems
+
+In an overdetermined system, there are more equations than unknowns. In general,
+this means that there is no exact solution, but there will be an approximate
+solution that is the closest fit to the equations. Overdetermined systems are
+common when the data is noisy, and the approximate solution is useful rather
+than giving up because there is no exact solution.
+
+*Thought this would work but it throws an error*
+
+from sympy import Symbol, nsolve
+
+x1 = Symbol('x1')
+
+x2 = Symbol('x2')
+
+f1 = 3 * x1**2 - 2 * x2**2 - 1
+
+f2 = x1**2 - 2 * x1 + x2**2 + 2 * x2 - 8
+
+f3 = x1 + x2
+
+print(nsolve((f1, f2, f3), (x1, x2), (-1, 1)))
+
+### Use SciPy on a lambda function *what is the advantage of this method?*
+
+You can *description*
+
+```py
+>>> from sympy.abc import x
+>>> from sympy.utilities.lambdify import implemented_function
+>>> from sympy import lambdify
+>>> from scipy import optimize
+>>> f = implemented_function('f', lambda x: x**3 - 1)
+>>> lam_f = lambdify(x, f(x))
+>>> sol = optimize.root_scalar(lam_f, bracket=[0, 2], method='brentq')
+>>> sol.root
+1.0
+```
+
+## Use the solution result
+
+### Substitute the result into an expression
+
+You can use [`subs`](sympy.core.basic.Basic.subs) to substitute numerical values
+into expressions to numerically evaluate them fully:
+
+```py
+>>> from sympy import cos, nsolve, Symbol, diff
+>>> x = Symbol('x')
+>>> f = cos(x) - x
+>>> x_value = nsolve(f, x, 1); x_value
+0.739085133215161
+>>> f.subs(x, x_value) # Verify that root is correct
+0
+>>> derivative = diff(f, x)
+>>> derivative.subs(x, x_value) # Calculate the derivative where the function value is zero
+-1.67361202918321
+```
+
+or leave some symbols as variables:
+
+```py
+>>> from sympy import cos, nsolve, Symbol, diff
+>>> x = Symbol('x')
+>>> f = cos(x) - x
+>>> x_value = nsolve(f, x, 1); x_value
+0.739085133215161
+>>> y = Symbol('y')
+>>> g = x * y**2
+>>> g.subs(x, x_value)
+0.739085133215161*y**2
+>>> derivative = diff(g, y)
+>>> derivative.subs(x, x_value)
+1.47817026643032*y
+```
+
+## Tradeoffs
+
+### Do not use `verify` for functions which are very steep near the root
+
+For functions which are very steep near the root, the verification of the
+solution may fail. In this case you should use the flag `verify=False` and
+independently verify the solution.
+
+```py
+>>> from sympy import cos, cosh, nsolve, Symbol
+>>> x = Symbol('x')
+>>> f = cos(x)*cosh(x) - 1
+>>> nsolve(f, 3.14*100)
+Traceback (most recent call last):
+...
+ValueError: Could not find root within given tolerance. (1.39267e+230 > 2.1684e-19)
+>>> ans = nsolve(f, 3.14*100, verify=False); ans
+312.588469032184
+>>> f.subs(x, ans).n(2)
+2.1e+121
+>>> (f/f.diff(x)).subs(x, ans).n(2)
+7.4e-15
```

Comment:
This is from the docstring. Do other people think this is worth discussing here, or is this too advanced for this guide? It's a bit subtle IMO, and really could use better explanation of what's going on.

Reply:
There are lots of numerical tricks to getting an equation in a well posed form for solving. Some general things that might be mentioned -- for which SymPy's symbolics can help -- would be good:
* working with the numerator: `n, d = eq.as_numer_denom(); nsolve(n, ...)`
* dividing by the derivative: `fd = eq/eq.diff(); nsolve(fd...)`
* removing large contants/common factors: `_eq = factor_terms(eq).as_coeff_Mul()[1]; nsolve(_eq...)`

But knowing how to get a root using `verify=False` when you know one is there but the routine is having troubles is a useful hint. But it can also be worked around (sometimes) by using some of the above methods:
```python
>>> f
cos(x)*cosh(x) - 1
>>> nsolve(f/f.diff(),312)
312.588469032184
```

Reply:
It would be good to give a little mathematical reasoning for these tricks. Without that, your first and second bullet points appear to contradict one another. 

The first point matters, if I understand correctly, if you get ZeroDivisionError. But it's not the only way that zero division can occur, so in some cases more advanced rewriting may be necessary.

The second bullet point is relevant when the root in question is a double root. Here's a simple example:

```
>>> nsolve(cos(x) - 1, 3)
-515.221191947442
>>> nsolve((cos(x) - 1)/-sin(x), 3)
6.28318530717931
```

The solution is correct (it's -164π), but it's not the closest one to the starting point. It's also impossible to use the bisection solver unless the root has odd multiplicity (it has to alternate signs at the endpoints of the interval).

```
>>> nsolve(cos(x) - 1, [1, 13], solver='bisect')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/aaronmeurer/Documents/Python/sympy/sympy/./sympy/utilities/decorator.py", line 88, in func_wrapper
    return func(*args, **kwargs)
  File "/Users/aaronmeurer/Documents/Python/sympy/sympy/./sympy/solvers/solvers.py", line 2979, in nsolve
  File "/Users/aaronmeurer/anaconda3/lib/python3.9/site-packages/mpmath/calculus/optimization.py", line 985, in findroot
    raise ValueError('Could not find root within given tolerance. '
ValueError: Could not find root within given tolerance. (0.211321969990149371762 > 2.16840434497100886801e-19)
Try another starting point or tweak arguments.
>>> nsolve((cos(x) - 1)/-sin(x), [1, 13], solver='bisect')
6.28318530717661
```

OTOH, this doesn't seem to always work

```
>>> nsolve(expand((x - 1)**4), 0)
0.998067087259351
>>> nsolve(expand((x - 1)**4)/diff((x - 1)**4, x).expand(), 0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/aaronmeurer/Documents/Python/sympy/sympy/./sympy/utilities/decorator.py", line 88, in func_wrapper
    return func(*args, **kwargs)
  File "/Users/aaronmeurer/Documents/Python/sympy/sympy/./sympy/solvers/solvers.py", line 2979, in nsolve
  File "/Users/aaronmeurer/anaconda3/lib/python3.9/site-packages/mpmath/calculus/optimization.py", line 969, in findroot
    for x, error in iterations:
  File "/Users/aaronmeurer/anaconda3/lib/python3.9/site-packages/mpmath/calculus/optimization.py", line 93, in __iter__
    f1 = f(x1)
  File "<lambdifygenerated-55>", line 2, in _lambdifygenerated
    return (x**4 - 4*x**3 + 6*x**2 - 4*x + 1)/(4*x**3 - 12*x**2 + 12*x - 4)
  File "<string>", line 7, in __div__
  File "/Users/aaronmeurer/anaconda3/lib/python3.9/site-packages/mpmath/libmp/libmpf.py", line 956, in mpf_div
    if t == fzero: raise ZeroDivisionError
ZeroDivisionError
```

In this case, using a different solver may produce better results, like `'mnewton'` which works better for multiple roots (mnewton also handles the cos(x) = 1 case well, so I'm unsure if it's ever better to use f/f' rather than just switching to mnewton).

For removing large constants, I agree that can be useful, especially if you can't get convergence, although the recipe you give might not always be correct (e.g., it might instead require doing a change of variables).

Reply:
Removed this section--I think it'd get too in-the-weeds for a user-friendly guide.